mec172xevb_assy6906: tests/kernel/device/ fails.

**Describe the bug**
Test `tests/kernel/device` fails.
```
 tests/kernel/device/kernel.device on mec172xevb_assy6906 error (Build failure)
 tests/kernel/device/kernel.device.minimallibc on mec172xevb_assy6906 error (Build failure)
 tests/kernel/device/kernel.device.pm on mec172xevb_assy6906 error (Build failure)
```

**To Reproduce**
`twister --hardware-map <path_to>/mec172x.map --device-testing -T tests/kernel/device/ -v`

**Expected behavior**
Pass.

**Impact**
Annoyance.

**Logs and console output**
```
INFO    - Using Ninja..
INFO    - Zephyr version: zephyr-v3.5.0-770-gaa3e79a88ac6
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...

Device testing on:

| Platform            |         ID | Serial device   |
|---------------------|------------|-----------------|
| mec172xevb_assy6906 | 0002451700 | /dev/ttyUSB0    |

INFO    - JOBS: 8
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 1/3 mec172xevb_assy6906       tests/kernel/device/kernel.device.pm                ERROR Build failure (device)
ERROR   - see: <path_to>/twister-out/mec172xevb_assy6906/tests/kernel/device/kernel.device.pm/build.log
INFO    - 2/3 mec172xevb_assy6906       tests/kernel/device/kernel.device                   ERROR Build failure (device)
ERROR   - see: <path_to>/twister-out/mec172xevb_assy6906/tests/kernel/device/kernel.device/build.log
INFO    - 3/3 mec172xevb_assy6906       tests/kernel/device/kernel.device.minimallibc       ERROR Build failure (device)
ERROR   - see: <path_to>/twister-out/mec172xevb_assy6906/tests/kernel/device/kernel.device.minimallibc/build.log

INFO    - 3 test scenarios (3 test instances) selected, 0 configurations skipped (0 by static filter, 0 at runtime).
INFO    - 0 of 3 test configurations passed (0.00%), 0 failed, 3 errored, 0 skipped with 0 warnings in 11.68 seconds
INFO    - In total 48 test cases were executed, 0 skipped on 1 out of total 630 platforms (0.16%)
INFO    - 0 test configurations executed on platforms, 3 test configurations were only built.

Hardware distribution summary:

| Board               |         ID |   Counter |
|---------------------|------------|-----------|
| mec172xevb_assy6906 | 0002451700 |         0 |
INFO    - Saving reports...
INFO    - Writing JSON report <path_to>twister-out/twister.json
INFO    - Writing xunit report <path_to>/twister-out/twister.xml...
INFO    - Writing xunit report <path_to>twister-out/twister_report.xml...
INFO    - -+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
INFO    - The following issues were found (showing the top 10 items):
INFO    - 1) tests/kernel/device/kernel.device on mec172xevb_assy6906 error (Build failure)
INFO    - 2) tests/kernel/device/kernel.device.minimallibc on mec172xevb_assy6906 error (Build failure)
INFO    - 3) tests/kernel/device/kernel.device.pm on mec172xevb_assy6906 error (Build failure)
INFO    -
INFO    - To rerun the tests, call twister using the following commandline:
INFO    - west twister -p <PLATFORM> -s <TEST ID>, for example:
INFO    -
INFO    - west twister -p mec172xevb_assy6906 -s tests/kernel/device/kernel.device.pm
INFO    - or with west:
INFO    - west build -p -b mec172xevb_assy6906 tests/kernel/device -T kernel.device.pm
INFO    - -+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
INFO    - Run completed
```

**Environment:**
 - OS: Linux
 - Toolchain Zephyr SDK v0.16.3
 - Commit https://github.com/zephyrproject-rtos/zephyr/commit/aa3e79a88ac635492c2baccf0f74dc26597dadf8

contents of
`<path_to>/twister-out/mec172xevb_assy6906/tests/kernel/device/kernel.device/build.log`
`<path_to>/twister-out/mec172xevb_assy6906/tests/kernel/device/kernel.device.pm/build.log`
`<path_to>/twister-out/mec172xevb_assy6906/tests/kernel/device/kernel.device.minimallibc/build.log`


```
FAILED: zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mchp_xec.c.obj
ccache <path_to>zephyr-sdk-0.16.3/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-DKERNEL -DPICOLIBC_INTEGER_PRINTF_SCANF -DTC_RUNID=8b0b9938b918d34a5f1d90fcbbcd3109
 -D_FORTIFY_SOURCE=1 -D_POSIX_C_SOURCE=200809 -D__LINUX_ERRNO_EXTENSIONS__ 
 -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I<path_to>/include 
 -I<path_to>/zephyr/twister-out/mec172xevb_assy6906/tests/kernel/device/kernel.device/zephyr/include/generated 
 -I<path_to>/zephyr/soc/arm/microchip_mec/mec172x -I/<paht_to>/zephyr/drivers 
 -I<path_to>/zephyr/soc/arm/microchip_mec/common/. 
 -I<path_to>/zephyr/subsys/testsuite/include 
 -I<path_to>/zephyr/subsys/testsuite/ztest/include 
 -I<path_to>/modules/hal/cmsis/CMSIS/Core/Include 
 -I<path_to>/zephyr/modules/cmsis/. -Wshadow -fno-strict-aliasing -Werror -Os 
 -imacros<path_to>/zephyr/twister-out/mec172xevb_assy6906/tests/kernel/device/kernel.device/zephyr/include/generated/autoconf.h 
 -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always 
 -mcpu=cortex-m4 -mthumb -mabi=aapcs -mfp16-format=ieee 
 --sysroot=<path_to>/zephyr-sdk-0.16.3/arm-zephyr-eabi/arm-zephyr-eabi 
 -imacros <path_to>/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat 
 -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith
  -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic 
  -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec 
  -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop 
  -fmacro-prefix-map=<path_to>/zephyr/tests/kernel/device=CMAKE_SOURCE_DIR 
  -fmacro-prefix-map=<path_to>/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=<path_to>zephyrproject=WEST_TOPDIR 
  -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD 
  -MT zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mchp_xec.c.obj 
  -MF zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mchp_xec.c.obj.d
   -o zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mchp_xec.c.obj 
   -c  <path_to>/zephyrproject/zephyr/drivers/serial/uart_mchp_xec.c
<path_to>/zephyrproject/zephyr/drivers/serial/uart_mchp_xec.c:230:13: error: 'uart_xec_pm_policy_state_lock_put' defined but not used [-Werror=unused-function]        
  230 | static void uart_xec_pm_policy_state_lock_put(enum uart_xec_pm_policy_state_flag flag)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<path_to>/zephyrproject/zephyr/drivers/serial/uart_mchp_xec.c:223:13: error: 'uart_xec_pm_policy_state_lock_get' defined but not used [-Werror=unused-function]        
  223 | static void uart_xec_pm_policy_state_lock_get(enum uart_xec_pm_policy_state_flag flag)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

```

--- a/drivers/serial/uart_mchp_xec.c
+++ b/drivers/serial/uart_mchp_xec.c
@@ -219,7 +219,7 @@ struct uart_xec_dev_data {
 
 static const struct uart_driver_api uart_xec_driver_api;
 
-#ifdef CONFIG_PM_DEVICE
+#if defined(CONFIG_PM_DEVICE) && defined(CONFIG_UART_CONSOLE_INPUT_EXPIRED)
 static void uart_xec_pm_policy_state_lock_get(enum uart_xec_pm_policy_state_flag flag)
 {
 	if (atomic_test_and_set_bit(pm_policy_state_flag, flag) == 0) {
@@ -644,7 +644,7 @@ static int uart_xec_fifo_fill(const struct device *dev, const uint8_t *tx_data,
 	k_spinlock_key_t key = k_spin_lock(&dev_data->lock);
 
 	for (i = 0; (i < size) && (regs->LSR & LSR_THRE) != 0; i++) {
-#ifdef CONFIG_PM_DEVICE
+#if defined(CONFIG_PM_DEVICE) && defined(CONFIG_UART_CONSOLE_INPUT_EXPIRED)
 		uart_xec_pm_policy_state_lock_get(UART_XEC_PM_POLICY_STATE_TX_FLAG);
 #endif
 		regs->RTXB = tx_data[i];
@@ -933,7 +933,7 @@ static void uart_xec_isr(const struct device *dev)
 		dev_data->cb(dev, dev_data->cb_data);
 	}
 
-#ifdef CONFIG_PM_DEVICE
+#if defined(CONFIG_PM_DEVICE) && defined(CONFIG_UART_CONSOLE_INPUT_EXPIRED)
 	if (uart_xec_irq_tx_complete(dev)) {
 		uart_xec_pm_policy_state_lock_put(UART_XEC_PM_POLICY_STATE_TX_FLAG);
 	}

